docs/modules/password_auth_provider_callbacks.md

```diff
@@ -166,6 +166,25 @@ any of the subsequent implementations of this callback. If every callback return
 the username provided by the user is used, if any (otherwise one is automatically
 generated).
 
+## `is_3pid_allowed`
+
+_First introduced in Synapse v1.53.0_
+
+```python
+async def is_3pid_allowed(self, medium: str, address: str, registration: bool) -> bool
+```
+
+Called when attempting to bind a third-party identifier (i.e. an email address or a phone
+number). The module is given the medium of the third-party identifier (which is `email` if
+the identifier is an email address, or `msisdn` if the identifier is a phone number) and
```

Comment:
We should probably note that new `medium`s could potentially be added in the future, so implementors should handle that? (It is somewhat unlikely, but :shrug: )

Reply:
I'll open an issue since that's not the only place we say `medium` can only have one of two values.

Reply:
Opened as https://github.com/matrix-org/synapse/issues/11935